src/pint/models/stand_alone_psr_binaries/BT_piecewise.py

```diff
@@ -0,0 +1,441 @@
+        T0X_per_toa = T0X_per_toa * u.d
+        A1X_per_toa = A1X_per_toa * ls
+        return [T0X_per_toa, A1X_per_toa]
+
+    def group_index(self):
+        index = []
+        for i in range(len(self._t)):
+            index1 = self.lower_index[i]
+            index2 = self.upper_index[i]
+            if index1 == index2:
+                index.append(index1)
+            else:
+                index.append(-1)
+        self.group_index_array = np.array(index)
+        return self.group_index_array
+
+    def toa_belongs_in_group(self, t):
+        group_no = []
+        lower_edge, upper_edge = self.get_group_boundaries()
+        for i in t:
+            # print(lower_edge)
+            lower_bound = np.searchsorted(lower_edge, i) - 1
+            upper_bound = np.searchsorted(upper_edge, i)
+            if lower_bound == upper_bound:
+                index_no = lower_bound
+            else:
+                index_no = -1
+
+            if index_no != -1:
+                group_no.append(self.piecewise_parameter_information[index_no][0])
+            else:
+                group_no.append(index_no)
+        # print(group_no)
+        return group_no
+
+    def get_group_boundaries(self):
+        lower_group_edge = []
+        upper_group_edge = []
+        # print("from get_group_boundaries")
+        if hasattr(self, "piecewise_parameter_information"):
+            for i in range(0, len(self.piecewise_parameter_information)):
+                lower_group_edge.append(
+                    self.piecewise_parameter_information[i][3].value
+                )
+                upper_group_edge.append(
+                    self.piecewise_parameter_information[i][4].value
+                )
+            return [lower_group_edge, upper_group_edge]
+
+    def a1(self):
+        self.A1_val = self.A1X_arr * ls
+        if hasattr(self, "A1X_per_toa") is True:
+            ret = self.A1X_per_toa + self.tt0 * self.A1DOT
+        else:
+            ret = self.A1 + self.tt0 * self.A1DOT
+        return ret
+
+    def get_tt0(self, barycentricTOA):
+        """tt0 = barycentricTOA - T0"""
+        if barycentricTOA is None or self.T0 is None:
+            tt0 = None
+            return tt0
+        if len(barycentricTOA) > 1:
+            if len(self.piecewise_parameter_information) > 0:
+                self.toa_belongs_in_group(
+                    barycentricTOA
+                )  # Defines object's index for each toa as an array of length = len(self._t)
+                (
+                    self.T0X_per_toa,
+                    self.A1X_per_toa,
+                ) = self.piecewise_parameter_from_information_array(
+                    self._t
+                )  # Uses the index for each toa array to create arrays where elements are the A1X/T0X to use with that toa
+        if len(barycentricTOA) > 1:
+            check = hasattr(self, "T0X_per_toa")
+            if hasattr(self, "T0X_per_toa") is True:
```

Comment:
If you're going to create `check`, just use `if check:`; but again, when might `T0X_per_toa` not be present?

Reply:
Testing variable, deleted.

Reply:
T0X_per_toa was not present on model initialisation, this was likely a hangover from testing when tt0 gets calculated